test/e2e/cypress/e2e/hosts_overview.cy.js

```diff
@@ -170,4 +170,42 @@ context('Hosts Overview', () => {
       });
     });
   });
+
+  describe('Clean-up', () => {
+    describe('Clean-up buttons should be visible only when needed', () => {
+      const heartbeatTargets = agents();
+      const failingHost = heartbeatTargets.shift();
+      before(() => {
+        cy.task('startAgentHeartbeat', heartbeatTargets);
+      });
+
+      it('should hide all Clean-up buttons when heartbeat starts except the first one', () => {
+        cy.get(
+          ':nth-child(1) > .w-48 > .bg-white > .text-jungle-green-500'
```

Comment:
What about this: `tr:nth-child(1) > td:nth-child(9)` where the 1st `1` is the row number.
The other query selector looks totally confusing